internal/gengapic/genrest.go

```diff
@@ -199,10 +198,11 @@ func (g *generator) emptyUnaryRESTCall(servName string, m *descriptor.MethodDesc
 	p("}")
 	p("")
 	p("httpRsp, err := client.Do(httpReq)")
-	p("defer httpRsp.Body.Close()")
 	p("if err != nil{")
 	p(" return err")
-	p("} else if httpRsp.StatusCode >= 400 {")
+	p("}")
+	p("defer httpRsp.Body.Close()")
+	p("if httpRsp.StatusCode != 200 {")
```

Comment:
nit: I would use `http.StatusOK` because it's more descriptive
https://golang.org/pkg/net/http/#pkg-constants